src/shared/vscode/commands2.ts

```diff
@@ -365,44 +369,192 @@ interface CommandInfo<T extends Callback> {
      * (default: 5 minutes)
      */
     readonly telemetryThrottleMs?: number | false
-}
-
-// This debouncing is a temporary safe-guard to mitigate the risk of instrumenting all
-// commands with telemetry at once. The alternative is to do a gradual rollout although
-// it's not much safer in terms of weeding out problematic commands.
-const telemetryInfo = new Map<string, { startTime: number; debounceCount: number }>()
 
-function getInstrumenter(id: string, threshold: number, telemetryName?: MetricName) {
-    const currentTime = Date.now()
-    const info = telemetryInfo.get(id)
+    /**
+     * The indexes of args in `execute()` that will be used to determine
+     * the uniqueness of the metric `vscode_executeCommand`.
+     * 
+     * - By default, `vscode_executeCommand` is throttled for metrics of the
+     *   same `id`.
+     * - Defining arg indexes for this property, the values of the args will be combined
+     *   with the `id`. This ensures throttling only happens for the same `id` + args
+     *   instead.
+     * - Indexes starts at 0.
+     *
+     * @example
+     * const argIndexes: ArgIndexes = {
+     *     0: undefined // the value is not important
```

Comment:
why do we need to support `undefined`? can we just require a name always?

Reply:
Current design is that setting a name will add it as a metric field. (eg "source").
When `undefined` we use that arg in the composite key but we do not include it as a metric field.

Reply:
Ok that's more specific than `value is not important` :)

We can leave this for now. But I'm starting to think that it would simplify a lot if we only defined compositeKey by metrics "metadata fields" instead of args. And with that constraint, if we want to discriminate `vscode_executeCommand` invocations, then yes, the "arg(s)" should (must) be stored _on the metric_. That simplifies everything.

Reply:
> We could always require a `string` and include all values at all indexes in the fields for `vscode_executeCommand` where the field name is the `string` and the value is the associated `arg

That sounds very good to me. It's one less "mapping" layer.